src/housekeeping.rs

```diff
@@ -5,7 +5,7 @@ use std::collections::{BTreeMap, BTreeSet};
 use std::path::Path;
 use tracing::{info, span, Level};
 
-pub type KnownViews = BTreeMap<String, BTreeSet<String>>;
+pub type KnownViews = BTreeMap<String, (git2::Oid, BTreeSet<String>)>;
```

Comment:
I think it would be more readable if it's not just a tuple but a struct with named fields, since it's not just used for return values but fields are referenced explicitly

Reply:
Even better would be two separate maps and moving this to use `lazy_static` like the other in process caches.
That will be some more work though. I think this needs some rework anyway, also to store filters as `Filter` instead of `String`.
So I would say let's merge it now to see if it has the desired impact and then do those refactorings, maybe on the weekend.

Reply:
I see, alright. But some tests currently fail 

![image](https://user-images.githubusercontent.com/2057191/162427208-08c7ba54-f8fa-490b-8d8b-27253565f728.png)